### DISCUSSION TITLE
Levensfasen - ObjectGepland met realisatiedatum (obv uitwerking 2)

### DISCUSSION THREAD

**Discussion by pmaria** [Geonovum/disgeo-imsor#38] [General]
Op basis van de discussie in #29 hebben we een diepgaandere uitwerking gemaakt die:
* Ondersteuning biedt voor simpele wijzigingen met toekomstmutaties
* Ondersteuning biedt voor meerdere plannen tegelijkertijd
* Ondersteuning biedt voor het bekijken van geplande objecten (icm met bestaande objecten) op een willekeurig moment in tijd.

**Comment by pmaria** [comment]
## Model

* expliciete relaties tussen `ObjectGepland`s en `Object`en
  * de relaties lopen van `ObjectGepland` naar `Object`. Daarmee verandert een object niet wanneer er een plan voor is.
  * de relatie `heeftGeleidTot` wordt alleen gelegd wanneer een plan daadwerkelijk leidt tot (een aanpassing van) een object.
* `realisatiedatum` op `ObjectGepland` om geplande toekomstsituatie te kunnen bekijken

![2-c-schaduw-gepland-obejcttype-relaties-realisatiedatum-model](https://github.com/Geonovum/disgeo-imsor/raw/master/issues/15-levensfasen/uitwerkingen/2-schaduw-gepland-objecttype/2-c-schaduw-gepland-obejcttype-relaties-realisatiedatum-model.png)

## Complex scenario met meerdere geplande objecten

![2-c-schaduw-gepland-obejcttype-relaties-realisatiedatum-complex](https://github.com/Geonovum/disgeo-imsor/raw/master/issues/15-levensfasen/uitwerkingen/2-schaduw-gepland-objecttype/2-c-schaduw-gepland-obejcttype-relaties-realisatiedatum-complex.png)

## Simpel scenario met toekomstmutaties, zonder geplande objecten

![2-c-schaduw-gepland-obejcttype-relaties-realisatiedatum-simpel](https://github.com/Geonovum/disgeo-imsor/raw/master/issues/15-levensfasen/uitwerkingen/2-schaduw-gepland-objecttype/2-c-schaduw-gepland-obejcttype-relaties-realisatiedatum-simpel.png)

**Comment by RiX012** [comment]
Op zich een duidelijk verhaal. Wel nog een aantal punten:

1. Deze optie moet valide gehouden worden met businessrules (bijvoorbeeld `status` `gesloopt`, kan niet relatie hebben met `status` `verbouw gepland`. Ik vind het lastig om dit niet in model te verwerken maar afhankelijk te laten zijn van de implementatie
2. Daarnaast is er de 'businessrule' dat datums op elkaar af moeten worden gestemd. Hoe houd je dat in de gaten?
3. Er zit ook nog de expliciete registratie van `versie` tussen, deze is ook afhankelijk van de datums.
4. Kortom ik vind de 'provenance' best lastig van dit model, maar ik weet niet zo goed of dat ook een argument bij de implementatie is
5. Ik vind nog steeds de semantiek van `ObjectGepland` ingewikkeld. Ik snap wel wat er bedoeld wordt maar of het nu echt een 'Gepland' object bevat (zoals de semantiek van de NEN2660-2 dat definieert vraag ik me af).
6. Ik vraag me ook tevens af over hoe dat gaat werken als een deel van een weg wordt aangepast. Dus je hebt een weg die loopt van punt A naar Z, daar heb je de geometrie van. Nu ga je een plan indienen om van D tot G de geometrie aan te passen (verbreding ofzo), als dat klaar is heb je dan een nieuwe geometrie van A naar Z of juist drie geometrie (A-C, D-G, G-Z) met elk aangepast registratie gegevens?

**Comment by RiX012** [comment]
Oja, volgens mij komt het overeen met de constructie Gepland vs Gerealiseerd van de NEN2660-2. Behalve de naamgeving? Misschien in lijn brengen?
